### ISSUE TITLE
Feature Request: When Central Monitor has not seen the Rig(s) for 10min, send a message

### ISSUE THREAD

**Issue by paulpoco** [RainbowMiner/RainbowMiner#420]
Feature Request: When Central Monitor has not seen the Rig(s) for 10min, send a message.

Could use Pushover?  https://pushover.net/api 
I use another app called Flexget that uses Pushover to notify.
![image](https://user-images.githubusercontent.com/801414/52539496-7b7ecb80-2d33-11e9-800a-96540e116a8c.png)

I just had my rig offline for 10hrs.  I was fine yesterday.  I updated today from .1 to .3 and that is when the PC crashing started, total black screen and cannot remote in.  I lower some of my Clocks just in case.

**Comment by paulpoco** [comment]
For now at least change the background of the website box that says Offline to red.
![screenshot_20190210-130546_chrome](https://user-images.githubusercontent.com/801414/52539612-2d6ac780-2d35-11e9-88ce-4f5ab5c30b63.png)

**Comment by RainbowMiner** [comment]
Nice! I will have a look into this.

**Comment by paulpoco** [comment]
@RainbowMiner here is my debug file, can you see anything that would cause the Rig to lock up?

Symptoms: Displays go blank, mouse gone, network gone as Teamviewer offline.

[debug-2019-02-11.zip](https://github.com/RainbowMiner/RainbowMiner/files/2851723/debug-2019-02-11.zip)

**Comment by RainbowMiner** [comment]
It looks like CryptoDredge starting for HMQ1725 has wrecked your system. The last entry in the log protocol has been

```
Starting miner (CryptoDredge-GPU#00-GPU#01): 'C:\Users\pjb\Desktop\RainbowMiner\Bin\NVIDIA-CryptoDredge\CryptoDredge.exe -r 10 -R 1 -b 127.0.0.1:31300 -d 0,1 -a hmq1725 --no-watchdog -o stratum+tcp://hmq1725.na.mine.zpool.ca:3747 -u XXX -p rainbowminer,c=DASH --log log_31300.txt --no-nvml'
```

So maybe better decrease overclocking for HMQ1725 for this miner.

**Comment by paulpoco** [comment]
I guess I need to learn how to use the MSI overclock info program and use Rainbowminer's OC feature?

Currently using ASUS OC program because my cards are ASUS.

**Comment by RainbowMiner** [comment]
Would you drop the following log file for further investigation?
```
C:\Users\pjb\Desktop\RainbowMiner\Bin\NVIDIA-CryptoDredge\log_31300.txt
```

**Comment by RainbowMiner** [comment]
RainbowMiner's OC feature is very handsome. You will not need Afterburner at all.

**Comment by paulpoco** [comment]
[log_31300.txt](https://github.com/RainbowMiner/RainbowMiner/files/2851900/log_31300.txt)

I lowered my OC a little, see screenshot.  I had it all the way to the right hand side since December 2017 without issue.
![screenshot_20190211-080753_teamviewer](https://user-images.githubusercontent.com/801414/52576327-b7b63880-2dd4-11e9-948d-8e4577dff187.jpg)

**Comment by paulpoco** [comment]
> RainbowMiner's OC feature is very handsome. You will not need Afterburner at all.

So I need the MSI program?  Not doing it right now so no hurry

**Comment by RainbowMiner** [comment]
No, you do not need any program except RainbowMiner. It can control all Nvidia OC functions.
I can prepare you a set of config files, you may use to start. Since I have GTX1070 rigs, this may work, out-of-the box.

**Comment by RainbowMiner** [comment]
[ocprofiles.config.txt](https://github.com/RainbowMiner/RainbowMiner/files/2852107/ocprofiles.config.txt)
[algorithms.config.txt](https://github.com/RainbowMiner/RainbowMiner/files/2852109/algorithms.config.txt)
[devices.config.txt](https://github.com/RainbowMiner/RainbowMiner/files/2852110/devices.config.txt)

- make a backup of your config files
- if this is a CPU/GTX1070 only rig, you can copy them into your Config folder.
- after that, edit config.txt and set `"EnableOCProfiles": "1",`

**Comment by paulpoco** [comment]
Ok thanks.  I have the ASUS 1070 8G OC models.  That why for over a year I had it maxed out, but I always wondered is the max memory clock helps or just uses more power.

So Rainbowminer will adjust fans to maintain below a certain temperature?

**Comment by paulpoco** [comment]
It hit close by accident.

I uninstall the ASUS afterburner?

**Comment by RainbowMiner** [comment]
No, let it be. It's helpful having this running for the fans and temperature control

**Comment by RainbowMiner** [comment]
Rainbowminer will set the oc but will not touch temp/fan for now.

**Comment by RainbowMiner** [comment]
If you use my config files, it will start with CPU/MEM +100/+400 for most of the algorithms.

```
Profile1: no oc (e.g. +0/+0)
Profile2: max. core / max. memory oc (e.g. +100/+400), good for most miners
Profile3: max. core / half memory oc (e.g. +100/+200), recm. for claymore ethash-keccak, neoscrypt
Profile4: max. core / low memory oc (e.g. +100/-500), recm. for lyra2re2, lyra2z, x16r, x16s
Profile5: max. core / reduced memory oc (e.g. +100/+350), recm. for claymore ethash-blake2s, claymore ethash-pascal
```

**Comment by paulpoco** [comment]
> Rainbowminer will set the oc but will not touch temp/fan for now.

Any special settings for Afterburner?  Just change clocks to defaults?

Change Fans to auto?  I currently just set fans to a fixed rate because I have found the auto setting does always work.

**Comment by RainbowMiner** [comment]
Use Afterburner and activate user controlled fan
![image](https://user-images.githubusercontent.com/39437538/52580251-443e1680-2e28-11e9-9ca1-241c0b549bf8.png)

**Comment by RainbowMiner** [comment]
So, set it to auto and use the user defined setup.

**Comment by RainbowMiner** [comment]
I have had very good results with this setup on ASUS 1070 8GB OC

**Comment by paulpoco** [comment]
Just went to update Asus gpuII and during install it needed directX 2010?  Weird

**Comment by RainbowMiner** [comment]
> Any special settings for Afterburner? Just change clocks to defaults?

You can leave it in any state. The only thing of importance is the setting of the fans. Afterburner will not show the current oc settings by the numbers, but you will see at the gauge that they are set.

**Comment by paulpoco** [comment]
The new version of ASUS Tweak2 has a "OC Scanner" feature now.
Somehow it scans the OC curve.
![20190211_162646](https://user-images.githubusercontent.com/801414/52603046-e4d90a00-2e19-11e9-9fcf-d5d2d89268ec.jpg)

**Comment by paulpoco** [comment]
reset the NVIDIA Benchmarks, see which Algo crashes.

**Comment by paulpoco** [comment]
Ok trying your config, I can see the different GPU clocks for different Algos, cool.

**Comment by giannibonfanti** [comment]
You can try my monitoring service, it has this feature already implemented.
It's integrated with RainbowMiner (thanks @RainbowMiner) and you can have different email notifications: when a rig goes online/offline and when a miner with low speed is detected.

If you need info I'm here to help, or on our support page.

www.brucomining.com

**Comment by giannibonfanti** [comment]
@paulpoco be sure also to add the notification types you prefer on your BrucoMining account settings page

**Comment by paulpoco** [comment]
@RainbowMiner I made a copy of my Rainbowminer to do the OC built in functions.  So I got the new folder running fine back a couple of weeks ago.  Well I just realized my start with Windows was still running the old folder.  So I would manually try and kill the running in the background Rainbowminer thinking I got it all.  Then start my new Rainbowminer folder.   Well I found it when I removed CPU mining from config.  I was still having high CPU usage.  Changed the start with Windows to the new folder.  I wonder why when running the "stopp" and it did nothing and then I has to end the mining processes.

Any way to start with Windows and have Rainbowminer in the foreground just as if you started it yourself with the run as admin?

**Comment by RainbowMiner** [comment]
Yes, of course. Use the windows scheduled task manager:

Manage → Scheduled Tasks → Create a Basic Task.
- Name it "RainbowMiner" → Next
- Select at logon → Next
- Select start program → Next
- Enter `cmd /c "C:\path-to-your-rainbowminer\Start.bat"`
- Enter `C:\path-to-your-rainbowminer` into start in → Next → Yes → Create (or similar)

Now find your newly created task in the scheduled task library and double click to edit.
- check "with highest privileges"
- goto Trigger and double click the only entry in the list: set "delay for" to 15 seconds → Ok
- goto Settings and uncheck "Stop a task if it runs longer than .."

That should work :)

**Comment by paulpoco** [comment]
@RainbowMiner that worked.  In Win 10 there are only 30sec,1min,etc options but that delay must cause it to start in the foreground.

Thanks.

**Comment by RainbowMiner** [comment]
Perfect :) I think you may overwrite the delay defaults by hand. Just directly type 15 seconds into the input field. It is input and select field.

**Comment by RainbowMiner** [comment]
eMail notifications are available in the current release v3.9.9.4
https://pushover.net notifications are available in the current developer master.zip (will be the next release v3.9.9.5)
